File: php/laravel_graphql/app/GraphQL/Type/UserType.php
<?php

namespace App\GraphQL\Type;
use GraphQL\Type\Definition\Type;
use Folklore\GraphQL\Support\Type as GraphQLType;

class UserType extends GraphQLType {

  protected $attributes = [
    'name' => 'User',
    'description' => 'A user'
  ];

  public function fields()
  {
    return [
      'id' => [
        'type' => Type::nonNull(Type::string()),
        'description' => 'The id of the user'
      ],
      'first_name' => [
        'type' => Type::nonNull(Type::string()),
        'description' => 'The first name of the user'
      ],
      'last_name' => [
        'type' => Type::nonNull(Type::string()),
        'description' => 'The last name of the user'
      ],
      'email' => [
        'type' => Type::string(),
        'description' => 'The email of user'
      ],
      'username' => [
        'type' => Type::nonNull(Type::string()),
        'description' => 'The username of the user'
      ],
    ];
  }

  protected function resolveEmailField($root, $args)
  {
    return strtolower($root->email);
  }

}

File: ruby/rails_graphql/app/api/types/user_type.rb
UserType = GraphQL::ObjectType.define do
  name "User"
  description "A user"

  interfaces [NodeIdentification.interface]
  global_id_field :id

  field :first_name, !types.String
  field :last_name, !types.String
  field :username, !types.String
  field :email, !types.String
  field :posts, types[!PostType]
  field :comments, types[!CommentType]
end
